src/Components/Moormanage/Technicians/Technicians.tsx

```diff
@@ -0,0 +1,200 @@
+  const [technicianRecord, setTechnicianRecord] = useState()
+  const [globalFilter, setGlobalFilter] = useState<string | null>(null)
+  const [selectedDate, setSelectedDate] = useState<Date | null>(new Date())
+  const [technicianData, setTechnicianData] = useState<TechnicianPayload[]>([])
+  const [filteredTechnicianData, setFilteredTechnicianData] = useState<TechnicianPayload[]>([])
+  const [getTechnicians] = useGetTechnicianMutation()
+
+  const [workOrderData, setWorkOrderData] = useState<BillsData[]>([
+    {
+      id: 0,
+      technician: 'Suncatcher',
+      techniciansName: 'John Smith',
+      dueDate: '3-12-2024',
+    },
+  ])
+
+  const technicianHeader = (
+    <div>
+      <div className="p-input-icon-left">
+        <i className="pi pi-search" />
+        <input
+          type="search"
+          value={globalFilter as string}
+          onChange={(e) => setGlobalFilter(e.target.value)}
+          placeholder="Search By name, Id..."
+          className="border-[1px] w-[30vw] p-2 pl-10 rounded-md"
+        />
+      </div>
+    </div>
+  )
+
+  const handleDateSelect = (date: Date) => {
+    setSelectedDate(date)
+  }
+
+  const handleDateUnselect = () => {
+    setSelectedDate(null)
+  }
+
+  const workOrder = (
+    <>
+      <div className="flex gap-40 rounded-md">
+        <div>
+          <p className="font-bold"> Work Orders </p>
+        </div>
+        <div>
+          <SelectButton value={value} onChange={(e) => setValue(e.value)} options={options} />
+        </div>
+      </div>
+    </>
+  )
+
+  const Billsheader = (
+    <div className="flex flex-wrap align-items-center gap-4 ">
+      <span className="text-sm font-bold text-[black]">Work Orders</span>
+      <div className=" ">
+        <div className="">
+          <SelectButton
+            style={{ fontSize: '0.2rem', fontWeight: 'bolder', height: '2rem' }}
+            value={value}
+            onChange={(e: SelectButtonChangeEvent) => setValue(e.value)}
+            options={options}
+          />
+        </div>
+      </div>
+
+      <div className="ml-72">
+        <Button
+          onClick={function (): void {
+            throw new Error('Function not implemented.')
+          }}
+          label={'Add New'}
+          style={{
+            backgroundColor: 'black',
+            height: '4vh',
+            fontSize: '0.75rem',
+            borderRadius: '0.20rem',
+          }}
+        />
+      </div>
+    </div>
+  )
+
+  const getTechniciansData = async () => {
+    await getTechnicians({})
+      .unwrap()
+      .then(async (response) => {
+        console.log('RESPONSE', response)
+        const { status, content } = response as TechnicianResponse
+        if (status === 200 && Array.isArray(content)) {
+          setTechnicianData(content)
+          setFilteredTechnicianData(content) // Initialize filtered data with all data
+        }
+      })
+  }
+
+  useEffect(() => {
+    getTechniciansData()
+  }, [])
```

Comment:
no dependency ?

Reply:
I want to ensure that this function runs only once, so we don't make multiple calls unnecessarily.

Reply:
This will lead to bugs (infinite rendering), 
create custom hook or memoize the function  and pass the memoized function as dep, so that useffect will be called only once.

Reply:
bit confused here , if possible can you explain it more or share any example